### PR TITLE
fixes and changelog

### PR DESCRIPTION
**Pull request by cliffordp**
[ET-787]

(This was discovered when implementing ETP shortcodes.)

image

[ET-787]: https://moderntribe.atlassian.net/browse/ET-787

### CODE REVIEW

File: src/views/blocks/attendees/view-link.php [hunk 1]
```diff
@@ -58,7 +59,7 @@
 
 $link = $view->get_tickets_page_url( $event_id, $is_event_page );
 
-$message = esc_html( sprintf( __( 'You have %1s for this %2s.', 'event-tickets' ), implode( _x( ' and ', 'separator if there are both RSVPs and Tickets', 'event-tickets' ), $counters ), $events_label_singular ) );
+$message = esc_html( sprintf( __( 'You have %1s for this %2s.', 'event-tickets' ), implode( _x( ' and ', 'separator if there are both RSVPs and Tickets', 'event-tickets' ), $counters ), $post_type_singular ) );
```
Comment (tr1b0t) [inline src/views/blocks/attendees/view-link.php:62]: * :no_entry_sign: A gettext call containing placeholders was found, but was not accompanied by a "translators:" comment on the line above to clarify the meaning of the placeholders.
* :no_entry_sign: Multiple placeholders should be ordered. Expected '%1$1s, %2$2s', but got %1s, %2s.

<sub>hash: 771e67bd957ff37f0161056466335f86</sub>

File: src/views/tickets/orders.php [hunk 2]
```diff
@@ -67,13 +68,15 @@
...
+$post_type_singular = $post_type ? $post_type->labels->singular_name : _x( 'Post', 'fallback post type singular name', 'event-tickets' );
+
 $is_event_page = class_exists( 'Tribe__Events__Main' ) && Tribe__Events__Main::POSTTYPE === $event->post_type;
 ?>
 
 <div id="tribe-events-content" class="tribe-events-single">
 	<p class="tribe-back">
 		<a href="<?php echo esc_url( get_permalink( $event_id ) ); ?>">
-			<?php printf( '&laquo; ' . esc_html__( 'View %s', 'event-tickets' ), $post_type->labels->singular_name ); ?>
+			<?php printf( '&laquo; ' . esc_html__( 'View %s', 'event-tickets' ), $post_type_singular ); ?>
```
Comment (tr1b0t) [inline src/views/tickets/orders.php:79]: * :no_entry_sign: A gettext call containing placeholders was found, but was not accompanied by a "translators:" comment on the line above to clarify the meaning of the placeholders.
* :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$post_type_singular'
* :no_entry_sign: All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$post_type_singular'.

<sub>hash: 862d49f8dc925284b2f7ca6c91aded8a</sub>

File: src/views/tickets/view-link.php [hunk 3]
```diff
@@ -52,7 +53,7 @@
 
 $link = $view->get_tickets_page_url( $event_id, $is_event_page );
 
-$message = esc_html( sprintf( __( 'You have %1s for this %2s.', 'event-tickets' ), implode( _x( ' and ', 'separator if there are both RSVPs and Tickets', 'event-tickets' ), $counters ), $events_label_singular ) );
+$message = esc_html( sprintf( __( 'You have %1s for this %2s.', 'event-tickets' ), implode( _x( ' and ', 'separator if there are both RSVPs and Tickets', 'event-tickets' ), $counters ), $post_type_singular ) );
```
Comment (tr1b0t) [inline src/views/tickets/view-link.php:56]: * :no_entry_sign: A gettext call containing placeholders was found, but was not accompanied by a "translators:" comment on the line above to clarify the meaning of the placeholders.
* :no_entry_sign: Multiple placeholders should be ordered. Expected '%1$1s, %2$2s', but got %1s, %2s.

<sub>hash: 63797602f09b5c453322381213ba3122</sub>

File: src/views/blocks/attendees/view-link.php [hunk 4]
```diff
@@ -58,7 +59,8 @@
 
 $link = $view->get_tickets_page_url( $event_id, $is_event_page );
 
-$message = esc_html( sprintf( __( 'You have %1s for this %2s.', 'event-tickets' ), implode( _x( ' and ', 'separator if there are both RSVPs and Tickets', 'event-tickets' ), $counters ), $events_label_singular ) );
+// Translators: 1: number of RSVPs and/or Tickets with accompanying ticket type text, 2: post type label
```
Comment (tr1b0t) [inline src/views/blocks/attendees/view-link.php:62]: :no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: 50cf0d1e80fdf010b903e5d9ebe77bf0</sub>

File: src/views/blocks/attendees/view-link.php [hunk 5]
```diff
@@ -58,7 +59,8 @@
...
+$message = esc_html( sprintf( __( 'You have %1s for this %2s.', 'event-tickets' ), implode( _x( ' and ', 'separator if there are both RSVPs and Tickets', 'event-tickets' ), $counters ), $post_type_singular ) );
```
Comment (tr1b0t) [inline src/views/blocks/attendees/view-link.php:63]: :no_entry_sign: Multiple placeholders should be ordered. Expected '%1$1s, %2$2s', but got %1s, %2s.

<sub>hash: d24ea5835aadae934c15db3ef7e9fb53</sub>

File: src/views/tickets/orders.php [hunk 6]
```diff
@@ -67,13 +68,18 @@
...
+
 $is_event_page = class_exists( 'Tribe__Events__Main' ) && Tribe__Events__Main::POSTTYPE === $event->post_type;
 ?>
 
 <div id="tribe-events-content" class="tribe-events-single">
 	<p class="tribe-back">
 		<a href="<?php echo esc_url( get_permalink( $event_id ) ); ?>">
-			<?php printf( '&laquo; ' . esc_html__( 'View %s', 'event-tickets' ), $post_type->labels->singular_name ); ?>
+			<?php
+			// Translators: post type label
```
Comment (tr1b0t) [inline src/views/tickets/orders.php:80]: :no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: 2902b78f23da794ce60e953af5116076</sub>

File: src/views/tickets/orders.php [hunk 7]
```diff
@@ -67,13 +68,18 @@
...
+			printf( '&laquo; ' . esc_html__( 'View %s', 'event-tickets' ), $post_type_singular );
```
Comment (tr1b0t) [inline src/views/tickets/orders.php:81]: * :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$post_type_singular'
* :no_entry_sign: All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$post_type_singular'.

<sub>hash: 4289f0e2cb991d3d955405b6ac1c28e2</sub>

File: src/views/tickets/view-link.php [hunk 8]
```diff
@@ -52,7 +53,8 @@
 
 $link = $view->get_tickets_page_url( $event_id, $is_event_page );
 
-$message = esc_html( sprintf( __( 'You have %1s for this %2s.', 'event-tickets' ), implode( _x( ' and ', 'separator if there are both RSVPs and Tickets', 'event-tickets' ), $counters ), $events_label_singular ) );
+// Translators: 1: number of RSVPs and/or Tickets with accompanying ticket type text, 2: post type label
```
Comment (tr1b0t) [inline src/views/tickets/view-link.php:56]: :no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: 9104891678a1195d86aa3d2f94bce389</sub>

File: src/views/tickets/view-link.php [hunk 9]
```diff
@@ -52,7 +53,8 @@
...
+$message = esc_html( sprintf( __( 'You have %1s for this %2s.', 'event-tickets' ), implode( _x( ' and ', 'separator if there are both RSVPs and Tickets', 'event-tickets' ), $counters ), $post_type_singular ) );
```
Comment (tr1b0t) [inline src/views/tickets/view-link.php:57]: :no_entry_sign: Multiple placeholders should be ordered. Expected '%1$1s, %2$2s', but got %1s, %2s.

<sub>hash: ac16235d75edacf675240ea770dab85f</sub>

File: src/views/tickets/orders-rsvp.php [hunk 10]
```diff
@@ -34,7 +37,7 @@
 			sprintf(
 				__( 'My %1$s for this %2$s', 'event-tickets' ),
```
Comment (juanfra) [inline src/views/tickets/orders-rsvp.php:38]: Let's add translators comments here.

File: src/views/tickets/orders.php [hunk 11]
```diff
@@ -67,13 +68,18 @@
...
+
 $is_event_page = class_exists( 'Tribe__Events__Main' ) && Tribe__Events__Main::POSTTYPE === $event->post_type;
 ?>
 
 <div id="tribe-events-content" class="tribe-events-single">
 	<p class="tribe-back">
 		<a href="<?php echo esc_url( get_permalink( $event_id ) ); ?>">
-			<?php printf( '&laquo; ' . esc_html__( 'View %s', 'event-tickets' ), $post_type->labels->singular_name ); ?>
+			<?php
+			// Translators: post type label.
```
Comment (juanfra) [inline src/views/tickets/orders.php:80]: Even if it's only one placeholder, I'd include it as part of the comments.

File: src/views/tickets/view-link.php [hunk 12]
```diff
@@ -29,21 +30,25 @@
...
-$rsvp_count   = $view->count_rsvp_attendees( $event_id, $user_id );
-$ticket_count = $view->count_ticket_attendees( $event_id, $user_id );
+$post_type_singular = $post_type ? $post_type->labels->singular_name : _x( 'Post', 'fallback post type singular name', 'event-tickets' );
+$counters           = [];
+$rsvp_count         = $view->count_rsvp_attendees( $event_id, $user_id );
+$ticket_count       = $view->count_ticket_attendees( $event_id, $user_id );
 
 if ( 1 === $rsvp_count ) {
-	$counters[] = sprintf( _x( '%1d %2s', 'RSVP count singular', 'event-tickets' ), $rsvp_count, tribe_get_rsvp_label_singular( basename( __FILE__ ) ) );
+	// Translators: 1: the number one, 2: singular RSVP label.
```
Comment (juanfra) [inline src/views/tickets/view-link.php:39]: Also for these, I'd use `%1d` and `%2s` in the comments. Translators are normally non tech people and maybe they need some literal stuff to interpret it.

Comment (cliffordp) [reply src/views/tickets/view-link.php:39]: See https://developer.wordpress.org/apis/handbook/internationalization/internationalization-guidelines/#variables

File: src/views/tickets/orders-rsvp.php (hunk 10 above)
Comment (cliffordp) [reply src/views/tickets/orders-rsvp.php:38]: good catch, and i found another as well.

**Pull request merged**